File: src/main/java/burp/util/BURPConfiguration.java
package burp.util;

import picocli.CommandLine;
import picocli.CommandLine.Command;
import picocli.CommandLine.Option;

@Command(name = "burp")
public class BURPConfiguration {

    @Option(names = {"-h", "--help"}, usageHelp = true, description = "Display a help message")
    public boolean help = false;

    @Option(names = {"-m", "--mappingFile"}, description = "The RML mapping file", required = true)
    public String mappingFile = null;

    @Option(names = {"-o", "--outputFile"}, description = "The output file")
    public String outputFile = null;

    @Option(names = {"-b", "--baseIRI"}, description = "Used in resolving relative IRIs produced by the RML mapping")
    public String baseIRI = null;

    @Option(names = "--skip-validation", description = "Skip shape validation")
    public boolean skipShapeValidation = false;

    @Option(names = "--allow-prefix-template", description = "Allow using turtle prefixes in templates expressions")
    public boolean allowPrefixInTemplate = false;

    @Option(names = "--dev", description = "Hot-reload mode for development (experimental)")
    public boolean dev = false;

    @Option(names = "--experimental-join", description = "Enable joins based on Indexes (experimental)")
    public boolean indexedJoins = false;

    public BURPConfiguration(String[] args) throws Exception {
        try {
            BURPConfiguration conf = CommandLine.populateCommand(this, args);
            if (conf.help) {
                new CommandLine(this).usage(System.out);
                System.exit(0);
            }

            if (conf.mappingFile == null) {
                throw new Exception("An RML mapping file is mandatory.");
            }
        } catch (CommandLine.ParameterException pe) {
            System.out.println(pe.getMessage());
            new CommandLine(this).usage(System.out);
            throw pe;
        }
    }
}

File: src/main/java/burp/util/BURPConfiguration.kt
package burp.util

import com.github.ajalt.clikt.completion.CompletionCandidates
import com.github.ajalt.clikt.core.CliktCommand
import com.github.ajalt.clikt.parameters.options.convert
import com.github.ajalt.clikt.parameters.options.default
import com.github.ajalt.clikt.parameters.options.option
import com.github.ajalt.clikt.parameters.options.required
import org.apache.jena.riot.Lang
import org.apache.jena.riot.RDFLanguages

class BURPConfiguration : CliktCommand(name = "burp") {

    val mappingFile: String by option(
        "-m", "--mappingFile",
        help = "The RML mapping file"
    ).required()

    val outputFile: String? by option(
        "-o", "--outputFile",
        help = "The output file"
    )

    val outputFormat: Lang? by option(
        "-f", "--outputFormat",
        help = "The format of the output file (default: deduced from output file)",
        completionCandidates = CompletionCandidates.Fixed(
            RDFLanguages.strLangNQuads, RDFLanguages.strLangTurtle, RDFLanguages.strLangTriG,
            RDFLanguages.strLangJSONLD, RDFLanguages.strLangRDFXML
        )
    ).convert { RDFLanguages.nameToLang(it) ?: fail("Unknown output format: $it") }

    val baseIRI: String by option(
        "-b", "--baseIRI",
        help = "Used in resolving relative IRIs produced by the RML mapping",
    ).default("http://example.org/")

    val reportFile: String? by option(
        "-r", "--reportFile",
        help = "The report file"
    )

    override fun run() {
        // We just use this command to parse the arguments
    }
}
